core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -468,6 +468,30 @@ trait EntityComponent {
       }
     }
 
+    //noinspection SqlDialectInspection
+    private object CheckForExistingEntityTypeQuery extends RawSqlQuery {
+      val driver: JdbcProfile = EntityComponent.this.driver
+
+      def doesEntityTypeAlreadyExist(workspaceContext: Workspace, entityType: String): ReadAction[Seq[Boolean]] = {
+
+        sql"""select exists (select name from ENTITY
+               where workspace_id=${workspaceContext.workspaceIdAsUUID} and entity_type = $entityType)
+          """.as[Boolean]
+      }
+    }
+
+    //noinspection SqlDialectInspection
+    private object ChangeEntityTypeNameQuery extends RawSqlQuery {
+      val driver: JdbcProfile = EntityComponent.this.driver
+
+      def changeEntityTypeName(workspaceContext: Workspace, oldName: String, newName: String): WriteAction[Int] = {
+
+        sqlu"""update ENTITY set entity_type = $newName
+              where workspace_id=${workspaceContext.workspaceIdAsUUID} and entity_type = $oldName
+          """
+      }
+    }
```

Comment:
Hm, what should *this* one do with soft-deleted entities? I think this should also include `and deleted=0` so it only renames active entities, but  I wonder if there's any reason to also rename all of them …

Reply:
I added the deleted=0 for consistency, but it seems like this would only matter in an edge case where we resuscitate deleted entities for a customer.